antelope-common/src/index.rs

```diff
@@ -1,19 +1,95 @@
 use std::collections::HashSet;
 
+use serde_json::Value;
 use substreams::pb::sf::substreams::index::v1::Keys;
-use substreams_antelope::pb::ActionTraces;
-
-use crate::actions::action_keys;
+use substreams_antelope::pb::{ActionTrace, ActionTraces, PermissionLevel};
 
 #[substreams::handlers::map]
 fn index_actions(actions: ActionTraces) -> Result<Keys, substreams::errors::Error> {
+    collect_keys(actions, action_keys)
+}
+
+#[substreams::handlers::map]
+fn index_actions_extra(actions: ActionTraces) -> Result<Keys, substreams::errors::Error> {
+    collect_keys(actions, action_keys_extra)
+}
+
+fn collect_keys<F>(
+    actions: ActionTraces,
+    key_extractor: F,
+) -> Result<Keys, substreams::errors::Error>
+where
+    F: Fn(&ActionTrace) -> Vec<String>,
+{
     let keys = actions
         .action_traces
-        .into_iter()
-        .flat_map(|action| action_keys(&action))
+        .iter()
+        .flat_map(key_extractor)
         .collect::<HashSet<_>>()
         .into_iter()
         .collect();
 
     Ok(Keys { keys })
 }
+
+// i.e. https://docs.dfuse.eosnation.io/eosio/public-apis/reference/search/terms/
+pub fn action_keys_extra(trace: &ActionTrace) -> Vec<String> {
+    let action = trace.action.as_ref().unwrap();
+    if action.account == "eosio" && action.name == "onblock" {
+        return vec![];
+    }
+    let mut keys = Vec::with_capacity(action.authorization.len() * 2 + 5 + 3);
+    let json_data: Value = match serde_json::from_str(&action.json_data) {
+        Ok(data) => data,
+        Err(_) => Value::Object(Default::default()),
+    };
+
+    keys.extend(vec![
+        format!("code:{}", action.account),
+        format!("receiver:{}", trace.receiver),
+        format!("action:{}", action.name),
+    ]);
+
+    keys.extend(
+        json_data
+            .as_object()
+            .unwrap()
```

Comment:
I think you should use `.expect(<message>)` here.

Reply:
Yeah expect would be better. Let's merge this as is since I already published new release and we are already using it. I'll add expect message for the next release.

Reply:
Np